docs/design/deployment_design.md

```diff
@@ -0,0 +1,199 @@
+
+### Targets
+
+- Mizar deployment for both k8s and Arktos.
+- Mizar deployment in different environment including local dev environment (one VM), GC  P environment (multiple VMs), AWS environment (multiple VMs).
+- Mizar deployment should support factors such as multi-apiserver, multi-etcd, multi-tenant
+- Support deploy two version of Mizar: for developer and for end user
+Mizar setup script should support update Mizar to new version.
+
+### Requirements
+
+- Mizar deployment should be easy for user. Only need one operation or one YAML file.
+- After deployment, user should be able to easily sanity check deployment result.
+
+### Definitions
+
+- **Mizar Phase 1**: current Mizar components are in the same cluster of k8s.
+
+  ![Simple Endpoint](png/mizar-phase-1.png)
+
+- **Mizar Phase 2**: Mizar is targeting to 2-cluster model. In the model, Mizar is running at a k8s or Arktos cluster, and it provides network functions to another k8s or Arktos cluster through Mizar proxy.
+
+  Mizar cluster should be able to run either k8s or Arktos.
+
+  ![Simple Endpoint](png/mizar-phase-2.png)
+
+We should produce same major deploy process for Mizar phase 1 and 2, while deploying different components.
+
+### Environments
+
+There are different environments that Mizar should be supporting.
+
+- k8s Kind environment
+
+  Typically k8s cluster can be launched by Kind. It's k8s in one VM, but with multiple nodes.
+
+  Mizar phase 2 requires two clusters. If Kind doesn't support deploy two clusters to one VM, we can use other means such as arktos-up to deploy one cluster other than cluster deployed by Kind.
+
+- k8s GCP environment
+
+  kubernetes running in Google Cloud Platform.
+
+- k8s AWS environment
+
+  kubernetes running in AWS.
+
+- arktos-up environment
+
+  Arktos running in one machine in one node mode. It's typical Arktos environment for dev and test.
+
+- Arktos GCP environment - single master
+
+  Arktos running in GCP. It has one master node and multiple worker nodes.
+
+- Arktos AWS environment - single master
+
+  Arktos running in AWS. It has one master node and multiple worker nodes.
+
+- Arktos GCP environment - multi master
+
+  Arktos running in GCP. It has multiple API servers and multiple ETCD. Consider it has multiple master nodes.
+
+- Arktos GCP environment with Kubemark
+
+  Arktos running in GCP. It uses Kubemark to simulate large clusters with hundreds of nodes. It's a good environment to demonstrate Mizar performance in large clusters.
+
+### Dev vs Prod of Mizar
+
+All the time these should be two version of Mizar to be deployed. One is dev version and one is prod version.
+
+- **Dev Version**, it always reflects latest code of Mizar. The code can be from any branch such as master, dev-next or any private branch. In this way, while developing Mizar, developers can easily test Mizar by latest code. It's for Mizar developers with dev or test purpose. 
```

Comment:
We shall add a continuous deployment into a test cluster for development. Probably use Jenkins or code deploy for that. Our CI/CD pipeline is currently broken, and this will help making it more robust.

Reply:
Updated.